test/src/org/labkey/test/tests/targetedms/TargetedMSChromatogramOptimizationTest.java

```diff
@@ -0,0 +1,140 @@
+{
+    private static final String SKY_FILE = "SmallMoleculeLibrary3.sky.zip";
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        TargetedMSChromatogramOptimizationTest init = (TargetedMSChromatogramOptimizationTest) getCurrentTest();
+        init.setupFolder(FolderType.Library);
+    }
+
+    @Override
+    protected String getProjectName()
+    {
+        return "TargetedMS Chromatogram Optimization Test";
+    }
+
+    @Test
+    public void testUpload() throws Exception
+    {
+        goToProjectHome();
+        importData(SKY_FILE);
+        clickAndWait(Locator.linkContainingText("Panorama Dashboard"));
+        File downloadedClibFile = doAndWaitForDownload(() -> clickButton("Download", 0));
+
+        log("Verifying table exists");
+        List<String> tablesToVerify = new LinkedList<String>(Arrays.asList("TransitionOptimization", "MoleculeTransitionOptimization", "Molecule", "MoleculeList", "MoleculePrecursor", "MoleculePrecursorRetentionTime"));
+        List<String> tablesNotPresent = tableExists(tablesToVerify, downloadedClibFile);
+        if (tablesNotPresent.size() != 0)
+            checker().verifyTrue("Some of the tables do not exists in SQLITE file" + Arrays.toString(tablesNotPresent.toArray()), false);
+
+        log("Verifying the SampleFile modifications");
+        checker().verifyTrue("Sample File does not have CePredictorId",
+                columnExists(downloadedClibFile, "SampleFile", "CePredictorId"));
+        checker().verifyTrue("Sample File does not have DpPredictorId",
+                columnExists(downloadedClibFile, "SampleFile", "DpPredictorId"));
+
+        log("Verifying the rows counts");
+        checker().verifyEquals("Invalid number of rows in transition optimization", getServerTableRowCount("TransitionOptimization"),
+                sizeOfTable(downloadedClibFile, "MoleculeTransitionOptimization"));
+        checker().verifyEquals("Invalid number of rows in Molecule", getServerTableRowCount("Molecule"),
+                sizeOfTable(downloadedClibFile, "Molecule"));
+        checker().verifyEquals("Invalid number of rows in MoleculePrecursor", getServerTableRowCount("MoleculePrecursor"),
+                sizeOfTable(downloadedClibFile, "MoleculePrecursor"));
+
+
+//        checker().verifyEquals("Invalid number of rows in MoleculePrecursorRetentionTime", getServerTableRowCount("MoleculePrecursorRetentionTime"),
+//                sizeOfTable(downloadedClibFile, "MoleculePrecursorRetentionTime"));
+//        checker().verifyEquals("Invalid number of rows in MoleculeList", getServerTableRowCount("MoleculeList"),
+//                sizeOfTable(downloadedClibFile, "MoleculeList"));
+
+
+    }
+
+    private int getServerTableRowCount(String tableName)
+    {
+        goToSchemaBrowser();
+        DataRegionTable table = viewQueryData("targetedms", tableName);
+        table.rowSelector().showAll();
+        try
+        {
+            table.goToView("LibraryMolecules");
```

Comment:
Seems like it would be cleaner to pass an optional viewName parameter to this method. If it's not null, the test would switch to that view.